include/boost/math/special_functions/hypergeometric_1F1.hpp

```diff
@@ -0,0 +1,207 @@
+
+///////////////////////////////////////////////////////////////////////////////
+//  Copyright 2014 Anton Bikineev
+//  Copyright 2014 Christopher Kormanyos
+//  Copyright 2014 John Maddock
+//  Copyright 2014 Paul Bristow
+//  Distributed under the Boost
+//  Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef _BOOST_HYPERGEOMETRIC_1F1_HPP_
+  #define _BOOST_HYPERGEOMETRIC_1F1_HPP_
+
+#include <boost/math/policies/policy.hpp>
+#include <boost/math/policies/error_handling.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_series.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_asym.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_rational.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_recurrence.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_pade.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_bessel.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_scaled_series.hpp>
+
+namespace boost { namespace math { namespace detail {
+
+   // check when 1f1 series can't decay to polynom
+   template <class T>
+   inline bool check_hypergeometric_1f1_parameters(const T& a, const T& b)
+   {
+      BOOST_MATH_STD_USING
+
+         if ((b <= 0) && (b == floor(b)))
+         {
+            if ((a >= 0) || (a < b) || (a != floor(a)))
+               return false;
+         }
+
+      return true;
+   }
+
+   template <class T, class Policy>
+   inline T hypergeometric_1f1_imp(const T& a, const T& b, const T& z, const Policy& pol)
+   {
+      BOOST_MATH_STD_USING // exp, fabs, sqrt
+
+         static const char* const function = "boost::math::hypergeometric_1f1<%1%,%1%,%1%>(%1%,%1%,%1%)";
+
+      if ((z == 0) || (a == 0))
+         return T(1);
+
+      // undefined result:
+      if (!detail::check_hypergeometric_1f1_parameters(a, b))
+         return policies::raise_domain_error<T>(
+            function,
+            "Function is indeterminate for negative integer b = %1%.",
+            b,
+            pol);
+
+      // other checks:
+      if (a == -1)
```

Comment:
Another branch off a set of measure zero.

Reply:
As above, force of habit.